### DISCUSSION TITLE
Kudos on the "cask unshallow" messaging

### DISCUSSION THREAD

**Discussion by apjanke** [Homebrew/discussions#347] [Casks]
Hey y'all,

I just went to `brew update` and got this message:

```
$  brew update; brew upgrade; brew cleanup
Error:
  homebrew-cask is a shallow clone.
To `brew update`, first run:
  git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask fetch --unshallow
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
automatically to avoid repeatedly performing an expensive unshallow operation in
CI systems (which should instead be fixed to not use shallow clones). Sorry for
the inconvenience!
```

Just wanted to give you props for going to the trouble to make a detailed and understandable message here explaining what was going on and why, with an easy copy-and-paste solution.

Keep up the good work!

Cheers,
Andrew

**Comment by wti** [comment]
Agreed, kudos!  

And the message could be improved by indicating that the operation might silently take a long time (if not also indicating roughly how much more disk space is required), and perhaps how to track progress.  That would avoid confusion for those who follow the instructions.  

Also, it might be more believable to those concerned about hacks if it indicated the date/version the change was made.

**Reply by gmw** [reply]
In my case, homebrew-core went from 153M to 420M and homebrew-cask from 240M to 443M, so roughly a 2-2.5x increase in storage.
